src/Runtime/DynMemRef.h

```diff
@@ -29,14 +29,34 @@ typedef int64_t INDEX_TYPE;
 // all ranks and type combinations.
 // We will refer to it as a DMR (Dynamic MemRef).
 struct DynMemRef {
+
```

Comment:
No offense, but Dynamic is super overloaded. From what I though you used it in the first time, you mean to have this as a representation of a constant array known in the runtime.

It almost feels like a "competitor" for dense element attribute, but without the attribute parts? Note that the dense element attribute does store data in packed form, but then wrap it in all sorts of attribute and ref thinks... not very intuitive. I recall you suggesting that we may use this at compile time also... then it would clearly be an alternative to dense element attribute.

Would be great if we can clearly position how this is different from other constructs. If we introduce new constructs that are similar to others.... we are only drifting from the common standard and make our code harder to read to others. 

So maybe you mean this as an array to support ONNX constant array at runtime? The output of our ONNX_MLIR presumably has no more ONNX types, so if its for runtime only, then is it a ONNXRuntimeConstantArray?